src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1374,8 +1374,18 @@ private function registerLockConfiguration(array $config, ContainerBuilder $cont
             foreach ($resourceStores as $storeDsn) {
                 $storeDsn = $container->resolveEnvPlaceholders($storeDsn, null, $usedEnvs);
                 switch (true) {
-                    case 'flock' === $storeDsn:
-                        $storeDefinition = new Reference('lock.store.flock');
+                    case 0 === strpos($storeDsn, 'flock'):
```

Comment:
`case 0 === strpos($storeDsn, 'flock://'):` + substr instead of str_replace below